plugins/global-search/src/utils/filter/AsyncProcessor.ts

```diff
@@ -24,11 +24,15 @@ export interface AsyncProcessorEvents<TOutput> extends EventMap {
 }
 
 function waitForIdle(): Promise<void> {
-    return new Promise(resolve =>
-        requestIdleCallback(() => {
-            resolve()
-        })
-    )
+    return new Promise(resolve => {
+        if ("requestIdleCallback" in window) {
+            requestIdleCallback(() => {
+                resolve()
+            })
+        } else {
+            setTimeout(resolve, 1)
+        }
```

Comment:
Nit: Nice! Maybe do this once at the top level? Not necessarily shim, just a `const`.